spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -171,7 +171,10 @@ public Filter[] pushFilters(Filter[] filters) {
         }
 
       } catch (Exception e) {
-        LOG.warn("Failed to check if {} can be pushed down: {}", filter, e.getMessage());
```

Comment:
I thought you wanted to log a message when a filter could not be converted above. This block is only for abnormal cases. I also wonder whether we should include the actual exception together with the trace. I kept the old logic when I touched it but I am not sure why we did not want to include a trace in the first place.

Reply:
Yea let me check this, i thought initially in those cases that we fail to bind, and get an exception.

Hence thought we should not log the stack, as that is a "normal" case.  

I think we had decided it was a bit polluting , in some discussion here:  https://github.com/apache/iceberg/pull/5254#issuecomment-1183804207  , but if it only for abnormal case , I agree we should have a stack.

Reply:
Yea I looked further, in the case that I was trying (a UDF column), it doesn't even get into Iceberg.  So we'd have to add the log in Spark side.  

in Iceberg, the exception case that I add the log for here is just for column which we dont support yet, like metadata column. 

I guess we can add a log when it doesnt match the entire partition as well, although not sure how to make correct phrasing, as iiuc we don't skip it but do evaluate it on lower levels?

Reply:
There is a check above to see if the converted filter is not null, that tells us if the filter is supported or not. We could log it there.

I am 50/50 on logging the trace. Let me think a bit.